### PR TITLE
Add open telemetry instrumentation guide

### PR DESCRIPTION
**Pull request by HuwONS**
### What

Adding the OT instrumentation guide to the guides section.

### How to review

Check the steps make sense and could be followed!

### Who can review

Anyone

### CODE REVIEW

File: guides/OT_INSTRUMENTATION.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+## Instrumenting http handlers
+There are a wide range of different facilities for instrumenting http calls. The simplest (taken here from dp-frontend-router) simply wraps a http handler (or Gorillamux router) with an opentelemetry handler:
+```
+httpHandler := router.New(routerConfig)
+otelHandler := otelhttp.NewHandler(httpHandler,"/")
+
+log.Info(ctx, "Starting server", log.Data{"config": cfg})
+
+s := &http.Server{
+Handler: otelHandler,
```
Comment (charlescolbourn) [inline guides/OT_INSTRUMENTATION.md:114]: could you add the indent here?

File: guides/OT_INSTRUMENTATION.md [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+
+A purley middleware approach can also be taken using Alice. Here you can see the instrumentation and logging middlewares applied:
+```
+func New(cfg Config) http.Handler {
+    router := mux.NewRouter()
+
+    middleware := alice.New(
+        dprequest.HandlerRequestID(16),
+        dpotelgo.OtelLoggingMiddleware,
```
Comment (charlescolbourn) [inline guides/OT_INSTRUMENTATION.md:170]: OtelLoggingMiddleware has been removed from the library now

File: guides/OT_INSTRUMENTATION.md [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+        serverError.Handler,
+        redirects.Handler,
+    )
+
+    newAlice := middleware.Then(router)
+}
+```
+
+
+A slightly different approach has to be taken if using a reverse proxy:
```
Comment (charlescolbourn) [inline guides/OT_INSTRUMENTATION.md:184]: lines 184-198 can probably be removed as the section "instrumenting http calls" covers it.

File: guides/OT_INSTRUMENTATION.md [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+
+# Instrumenting Go services for OT
+
+The following environment variables need to be set:
+```
+OTEL_SERVICE_NAME=<service name>
+OTEL_EXPORTER_OTLP_ENDPOINT=localhost:4317
+```
+
+
```
[the remark this answers is not in the document]
Comment (HuwONS) [reply guides/OT_INSTRUMENTATION.md:81]: Is this also the case for the java guide?

**Review by franmoore05 — APPROVED** [review]

**Pull request merged**